sdcm/provision/gce/virtual_machine_provider.py

```diff
@@ -0,0 +1,140 @@
+import logging
+import random
+from typing import Type
+
+from libcloud.compute.base import Node
+from libcloud.compute.types import NodeState
+
+from sdcm.keystore import KeyStore
+from sdcm.provision.gce.disk_struct_provider import DiskStructProvider, DiskStructArgs
+from sdcm.provision.gce.metadata_provider import MetadataProvider, MetadataArgs
+from sdcm.provision.provisioner import InstanceDefinition, PricingModel
+from sdcm.utils.gce_utils import get_gce_service
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class VMInstanceAlreadyExists(Exception):
+    ...
+
+
+class VMInstanceDestructionUnsuccessful(Exception):
+    ...
+
+
+class VMInstanceRebootUnsuccessful(Exception):
+    ...
+
+
+class VMProvisionError(Exception):
+    ...
+
+
+class VirtualMachineProvider:
+    def __init__(self, region: str,
+                 disk_struct_provider: Type[DiskStructProvider],
+                 metadata_provider: Type[MetadataProvider]):
+        self._region = region
+        self._gce_service = get_gce_service(region=region)
+        self._location = self._get_random_location_in_region()
+        self._service_accounts = KeyStore().get_gcp_service_accounts()
```

Comment:
can we get it from `_gce_service`?

Reply:
It looks like we can't. We can get the e-mail portion, but not the scopes that we get from `get_gcp_service_accounts()`

Reply:
Then we should prefer providing it as constructor parameter rather than create dependency to `KeyStore`.